src/views/shared/amount-input/amount-input.view.jsx

```diff
@@ -25,9 +25,19 @@ function AmountInput(Component) {
     const [isAmountCompressedInvalid, setIsAmountCompressedInvalid] = React.useState(undefined);
 
     React.useEffect(() => {
+      let isActive = true;
+
       getProvider()
         .getGasPrice()
-        .then((gasPrice) => setGasPrice(gasPrice));
+        .then((gasPrice) => {
+          if (isActive) {
+            setGasPrice(gasPrice);
+          }
+        });
+
+      return () => {
+        isActive = false;
```

Comment:
I don't get this, what is it for?

Reply:
Indeed, it's ugly but has been added to prevent [this error](https://github.com/hermeznetwork/wallet-ui/pull/668#issuecomment-929298899) which was occurring because we were attempting to set a state on an unmounted component. We should definitely find a better approach than this mutable flag. More info [here](https://0xpolygon.slack.com/archives/C02FS3PHEGH/p1632840307005600).

Reply:
I'm no React expert. What other options do we have?

Reply:
I'm refactoring for other pr and I'm going to remove this effect

Reply:
Do you mean you are going to remove this in a different PR? Which one?

Reply:
My last change in https://github.com/hermeznetwork/wallet-ui/pull/665 removes that effect

Reply:
So can it be removed here then?

Reply:
it's going to create a conflict so I'm waiting to close the other pr to solve the conflict here